From 9e0d0e65261384bde8cb7f1932738bbeb19d3292 Mon Sep 17 00:00:00 2001
From: Carlo Cabrera <30379873+carlocab@users.noreply.github.com>
Date: Thu, 1 Sep 2022 13:18:39 +0800
Subject: [PATCH] Fix `sed` invocation.

The current `sed` call works only for GNU sed. We can fix that by using
the `-e` flag.
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ec36673d7..b85add059 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -542,7 +542,7 @@ if (GPERF AND SED AND RUN_GPERF_IF_PRESENT)
         add_custom_command(OUTPUT poppler/${input}.c
                        COMMAND ${GPERF} poppler/${input}.gperf > ${CMAKE_CURRENT_BINARY_DIR}/poppler/${input}.c
                        COMMAND ${GPERF} poppler/${input}.gperf > ${CMAKE_CURRENT_SOURCE_DIR}/poppler/${input}.pregenerated.c
-                       COMMAND ${SED} -i "s#${GPERF}#gperf#" ${CMAKE_CURRENT_SOURCE_DIR}/poppler/${input}.pregenerated.c
+                       COMMAND ${SED} -i -e "s#${GPERF}#gperf#" ${CMAKE_CURRENT_SOURCE_DIR}/poppler/${input}.pregenerated.c
                        COMMAND clang-format -i ${CMAKE_CURRENT_SOURCE_DIR}/poppler/${input}.pregenerated.c || true
                        DEPENDS poppler/${input}.gperf
                        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
-- 
GitLab

